txtorcon/endpoints.py

```diff
@@ -667,6 +668,7 @@ class TorClientEndpoint(object):
     socks_ports_to_try = [9050, 9150]
 
     def __init__(self, host, port,
+                 socks_socket=None,
                  socks_hostname=None, socks_port=None,
```

Comment:
ok i've turn the new argument into socks_endpoint... but i did not yet try to change all the default things to endpoints.